src/vs/workbench/contrib/extensions/electron-sandbox/remoteExtensionsInit.ts

```diff
@@ -26,17 +29,74 @@ import { IExtensionManagementServerService } from '../../../services/extensionMa
 import { IExtensionManifestPropertiesService } from '../../../services/extensions/common/extensionManifestPropertiesService.js';
 import { IRemoteAgentService } from '../../../services/remote/common/remoteAgentService.js';
 
-export class InstallFailedRemoteExtensionsContribution implements IWorkbenchContribution {
+export class InstallRemoteExtensionsContribution implements IWorkbenchContribution {
 	constructor(
 		@IRemoteAgentService private readonly remoteAgentService: IRemoteAgentService,
 		@IRemoteExtensionsScannerService private readonly remoteExtensionsScannerService: IRemoteExtensionsScannerService,
 		@IExtensionGalleryService private readonly extensionGalleryService: IExtensionGalleryService,
 		@IExtensionManagementServerService private readonly extensionManagementServerService: IExtensionManagementServerService,
-		@ILogService private readonly logService: ILogService
+		@ILogService private readonly logService: ILogService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IProductService private readonly productService: IProductService,
 	) {
+		this.installDefaultRemoteExtensions();
 		this.installFailedRemoteExtensions();
 	}
 
+	private async installDefaultRemoteExtensions(): Promise<void> {
+		if (!this.remoteAgentService.getConnection()) {
+			return;
+		}
+
+		if (!this.extensionManagementServerService.remoteExtensionManagementServer) {
+			this.logService.error('No remote extension management server available');
+			return;
+		}
+
+		const settingValue = this.configurationService.getValue<string[]>(REMOTE_DEFAULT_EXTENSIONS);
+		if (!settingValue.length) {
```

Comment:
Is it possible for settingValue to be null or undefined? If so, `settingValue?`

Reply:
I believe what I have is safe but an extra `?` won't hurt